machineconfiguration/v1/types_machineosbuild.go

```diff
@@ -0,0 +1,198 @@
+
+	// items contains a collection of MachineOSBuild resources.
+	Items []MachineOSBuild `json:"items"`
+}
+
+// MachineOSBuildSpec describes information about a build process primarily populated from a MachineOSConfig object.
+type MachineOSBuildSpec struct {
+	// desiredConfig is the desired config we want to build an image for.
+	// +kubebuilder:validation:Required
+	DesiredConfig RenderedMachineConfigReference `json:"desiredConfig"`
+	// machineOSConfig is the config object which the build is based off of
+	// +kubebuilder:validation:Required
+	MachineOSConfig MachineOSConfigReference `json:"machineOSConfig"`
+	// renderedImagePushspec is set from the MachineOSConfig
+	// The format of the image pullspec is:
+	// host[:port][/namespace]/name:<tag> or svc_name.namespace.svc[:port]/repository/name:<tag>
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=447
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^([a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+(:[0-9]{2,5})?(/[a-zA-Z0-9-_]{1,61})*/[a-zA-Z0-9-_.]+:[a-zA-Z0-9._-]+$') || self.matches('^[^.]+\\.[^.]+\\.svc:\\d+\\/[^\\/]+\\/[^\\/]+:[^\\/]+$')`,message="the OCI Image name should follow the host[:port][/namespace]/name format, resembling a valid URL without the scheme. Or it must be a valid .svc followed by a port, repository, image name, and tag."
+	// +kubebuilder:validation:Required
+	RenderedImagePushspec string `json:"renderedImagePushspec"`
+}
+
+// MachineOSBuildStatus describes the state of a build and other helpful information.
+type MachineOSBuildStatus struct {
+	// conditions are state related conditions for the build. Valid types are:
+	// Prepared, Building, Failed, Interrupted, and Succeeded.
+	// Once a Build is marked as Failed or Interrupted, no future conditions can be set.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:validation:XValidation:rule="self.exists(x, x.type == 'Failed') ? self == oldSelf : true",message="once a Failed condition is set, conditions are immutable"
+	// +kubebuilder:validation:XValidation:rule="self.exists(x, x.type == 'Interrupted') ? self == oldSelf : true",message="once an Interrupted condition is set, conditions are immutable"
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// BuilderReference describes which ImageBuilder backend to use for this build
+	// +optional
+	BuilderReference *MachineOSBuilderReference `json:"builderReference"`
+	// relatedObjects is a list of references to ephemeral objects such as ConfigMaps or Secrets that are meant to be consumed while the build process runs.
+	// After a successful build or when this MachineOSBuild is deleted, these ephemeral objects should be deleted.
+	// However, in the event of a failed build, the objects will not be deleted to allow for inspection and debugging of the failed build process.
```

Comment:
Will they be deleted when the MachineOSBuild is deleted? Should they always just be deleted when the MachineOSBuild is deleted?